src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,374 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "TemporarySummon.h"
+
+enum ThalorienDawnseekerTexts
+{
+    SAY_THALORIEN_1 = 0,
+    SAY_THALORIEN_2 = 1,
+    SAY_THALORIEN_3 = 2,
+    SAY_THALORIEN_4 = 3,
+    SAY_THALORIEN_5 = 4,
+    SAY_THALORIEN_6 = 5,
+    SAY_THALORIEN_7 = 6,
+    SAY_THALORIEN_8 = 7,
+    SAY_THALORIEN_9 = 8,
+    SAY_THALORIEN_10 = 9,
+
+    SAY_MORLEN_1 = 0,
+    SAY_MORLEN_2 = 1,
+    SAY_MORLEN_3 = 2,
+    SAY_MORLEN_4 = 3,
+    SAY_MORLEN_5 = 4
+};
+
+enum ThalorienDawnseekerSummons
+{
+    SUMMON_SUNWELL_DEFENDER = 1,
+    SUMMON_SCOURGE_ZOMBIE   = 2,
+    SUMMON_GHOUL_INVADER    = 3,
+    SUMMON_CRYPT_RAIDER     = 4
+};
+
+enum ThalorienDawnseekerActions
+{
+    ACTION_START_QUEST = 1
+};
+
+enum ThalorienDawnseekerEvents
+{
+    EVENT_INTRO_0       = 1,
+    EVENT_INTRO_1       = 2,
+    EVENT_INTRO_2       = 3,
+    EVENT_INTRO_3       = 4,
+    EVENT_SALUTE        = 5,
+    EVENT_DEFENDERS_RUN = 6,
+    EVENT_THALORIEN_GO  = 7,
+    EVENT_INTRO_4       = 8,
+    EVENT_SUMMON_MORLEN = 9,
+    EVENT_INTRO_5       = 10,
+    EVENT_MORLEN_1      = 11,
+    EVENT_SPAWN_WAVE_1  = 12,
+    EVENT_SPAWN_WAVE_2  = 13,
+    EVENT_SPAWN_WAVE_3  = 14,
+    EVENT_WAVE_ATTACK   = 15,
+    EVENT_MORLEN_2      = 16,
+    EVENT_MORLEN_ATTACK = 17,
+    EVENT_OUTRO_1       = 18,
+    EVENT_OUTRO_2       = 19,
+    EVENT_OUTRO_3       = 20,
+    EVENT_OUTRO_4       = 21,
+    EVENT_KNEEL         = 22
+};
+
+enum ThalorienDawnseeker
+{
+    MENU_EXAMINE_REMAINS        = 10908,
+
+    NPC_THALORIEN_DAWNSEEKER    = 37205,
+    NPC_SUNWELL_DEFENDER        = 37211,
+    NPC_MORLEN_GOLDGRIP         = 37542,
+    NPC_SCOURGE_ZOMBIE          = 37538,
+    NPC_GHOUL_INVADER           = 37539,
+    NPC_CRYPT_RAIDER            = 37541,
+
+    SPELL_BLOOD_PRESENCE        = 50689,
+    SPELL_KILL_CREDIT           = 70265,
+    SPELL_POLYMORPH_VISUAL      = 27123
+};
+
+Position const thalorienSummon = { 11795.32f, -7070.476f, 26.27511f, 5.67232f  };
+Position const thalorienFight = { 11788.46f, -7063.375f, 25.79677f, 3.054326f };
+Position const morlenSummon = { 11766.46f, -7050.078f, 26.19846f, 5.637414f  };
+
+class npc_thalorien_dawnseeker : public CreatureScript
+{
+public:
+    npc_thalorien_dawnseeker() : CreatureScript("npc_thalorien_dawnseeker") { }
+
+    struct npc_thalorien_dawnseekerAI : public ScriptedAI
+    {
+        npc_thalorien_dawnseekerAI(Creature* creature) : ScriptedAI(creature), _summons(me) { }
+
+        bool GossipSelect(Player* player, uint32 menuId, uint32 /*gossipListId*/) override
+        {
+            if (menuId == MENU_EXAMINE_REMAINS)
```

Comment:
What if two players have gossip window open? and both click it

Reply:
After first one clicks, menu is closed and event begins.

Reply:
Is it closed for the other player tho?

Reply:
Closed for both meaning you won't be able to start it again shortly.